### DISCUSSION TITLE
Questions about Labeling Options and Faceting in Compositional Barplots

### DISCUSSION THREAD

**Discussion by timz0605** [david-barnett/microViz#172] [General]
Hello David!

I've been making some progress in creating a barplot to show the composition of taxa in my samples. But I've been having some small troubles adjusting the aesthetics and would appreciate your input!

```
q <- ps_fraction %>% 
  tax_fix() %>% 
  comp_barplot(tax_level = "phylum", 
               n_taxa = 15, 
               merge_other = TRUE, 
               tax_order = prev, 
               bar_width = 0.9, 
               bar_outline_colour = NA,
               label = "abbreviationLabel",
               sample_order = desired_order) +
  coord_flip() +
  facet_wrap("sizeFraction", nrow = 1, scales = "free") +
  labs(x = "Individual ARMS by Location", 
       y = "Relative Abundance") +
  guides(fill = guide_legend(title = "Phylum")) +
  theme(
 ...... (some commands here making the figure look pretty)
  )
q
```
And below is what I am getting
![image](https://github.com/user-attachments/assets/2f6edef4-c0ad-4aa6-a1fc-165573c2296a)

Below are my questions:

1/ I have been trying to manually set the texts of facet strips, but cannot find a way to do it. For example, maybe changing the "106nm" to "106 nm Fraction" or "Small Fraction". I wonder if you have any ideas for achieving this?

2/ Right now, all bars in the facets have labels next to them. Is there a way to only show the labels on one side and for once (either left or right), so that the figure is easier to read?

3/ This is a tricky part and I am still thinking about it. As you notice, for the 500nm facet, the result from CLP-1 is missing (due to sequencing error), so there are only 11 bars compared to the sessile which has 12 bars. In this case, is there still a way to align the bars? Maybe left the CLP-1 as blank? (this might be very tricky, and it's totally fine if too much work is involved)

Again, thank you so much and I appreciate your help!

**Comment by david-barnett** [comment]
Quick thoughts:

1. You can use ps_mutate to change the levels of the sizeFraction variable to the labels you want for the facets.
2. You can use the x argument of comp_barplot and "fixed" scale on the facetting (similar to what's shown here: <https://david-barnett.github.io/microViz/articles/web-only/compositions.html#faceting-by-timepoint-and-another-variable>
3. would be solved by point 2

**Comment by timz0605** [comment]
Yes, now it works perfectly. Thank you so much!
